rlgym_tools/sb3_utils/sb3_log_reward.py

```diff
@@ -1,44 +1,152 @@
+    Simple reward function for logging individual rewards to a custom Logger.
+    """
+    def __init__(self, logger: Logger, reward_function: RewardFunction):
+        super().__init__()
+        self.logger = logger
+        self.reward_function = reward_function
+        self.reward_sum = 0
+        self.episode_steps = 0
+        self.global_steps = 0
+
+    def reset(self, initial_state: GameState):
+        if self.episode_steps > 0:
+            rew_fn_type = type(self.reward_function)
+            mean_reward = self.reward_sum / self.episode_steps
+            if rew_fn_type.__str__ is not object.__str__:
+                self.logger.record(f"{self.reward_function}/ep_rew_mean", mean_reward)
+            else:
+                self.logger.record(f"{rew_fn_type.__name__}/ep_rew_mean", mean_reward)
+            self.logger.dump(self.global_steps)
+            self.reward_sum = 0
+            self.episode_steps = 0
+            self.global_steps += 1
+
+        self.reward_function.reset(initial_state)
+
+    def get_reward(self, player: PlayerData, state: GameState, previous_action: np.ndarray) -> float:
+        rew = self.reward_function.get_reward(player, state, previous_action)
+        self.reward_sum += rew
+        self.episode_steps += 1
+        return rew
+
+    def get_final_reward(self, player: PlayerData, state: GameState, previous_action: np.ndarray) -> float:
+        rew = self.reward_function.get_final_reward(player, state, previous_action)
+        self.reward_sum += rew
+        self.episode_steps += 1
+        return rew
+
+class SB3CombinedLogReward(CombinedReward):
+
+    index_tracker = 0 # So we can uniquely identify each instance
+
+    def __init__(
+            self,
+            reward_functions: Tuple[RewardFunction, ...],
+            reward_weights: Optional[Tuple[float, ...]] = None
+    ):
+        """
+        Creates the combined reward using multiple rewards, and a potential set
+        of weights for each reward. Will also log the weighted rewards to
+        the model's logger if a SB3CombinedLogRewardCallback is provided to the
+        learner.
+
+        :param reward_functions: Each individual reward function.
+        :param reward_weights: The weights for each reward.
+        """
+        super().__init__(reward_functions,reward_weights)
+        self.index = SB3CombinedLogReward.index_tracker
+        SB3CombinedLogReward.index_tracker += 1
+
+        # make sure there is a folder to dump to
+        os.makedirs('bin/combinedlogfiles', exist_ok=True)
+        self.dumpfile = f'bin/combinedlogfiles/{self.index}.txt'
+
+        # clean out any files that are left over from previous runs
+        if self.index == 0:
+            for file in os.listdir('bin/combinedlogfiles'):
+                if '.txt' in file:
+                    os.remove('bin/combinedlogfiles/'+file)
+
+
+        # initiates the array that will store the rewards
+        self.returns = np.zeros(len(self.reward_functions))

```

Comment:
should this be a numpy array? Shouldn't this be a list given how many appends/concatenates there are below.